cfme/tests/infrastructure/test_providers.py

```diff
@@ -523,3 +529,55 @@ def test_compare_templates(appliance, setup_provider_min_templates, provider, mi
 
     t_list = [t.name for t in t_coll]
     assert compare_view.verify_checked_items_compared(t_list, compare_view)
+
+
+@pytest.fixture
+@pytest.mark.provider([AnsibleTowerProvider], selector=ONE_PER_VERSION, scope="function")
+def provider_with_special_characters(provider):
+    # The AWX version 3.4 cannot be patched.
+
+    template_name = 'add_special_characters'
+    session = requests.session()
+    session.verify = False
+
+    extra_vars_with_special_chars = {'extra_vars': dedent(
+        r"""
+        ---
+        entry_path: "test\/21\\test"
+        """)}
+    empty_extra_vars = {'extra_vars': ""}
+
+    with closing(session):
+        url = provider.url
+        creds = credentials[provider.data.credentials]
+        session.auth = (creds['username'], creds['password'])
+        response = session.get(f'{url}/job_templates/', params=dict(name=template_name))
+        assert response.ok
+        template, = response.json()['results']
+
+        response = session.patch(f'{url}/job_templates/{template["id"]}/',
+                                 json=extra_vars_with_special_chars)
+        assert response.ok
+
+        yield provider
+
+        response = session.patch(f'{url}/job_templates/{template["id"]}/',
+                                 json=empty_extra_vars)
+        assert response.ok
+
+
+@pytest.mark.meta(automates=[1819310])
+@pytest.mark.provider([AnsibleTowerProvider], selector=ONE_PER_VERSION, scope="function")
+@pytest.mark.uncollectif(lambda provider: provider.version == Version(3.4),
+                         reason="The AWX version 3.4 cannot be modified to contain special chars.")
+def test_awx_with_special_chars_refreshes(provider_with_special_characters: AnsibleTowerProvider):
+    """
+    Tests whether provider with special characters in template definition refreshes in CFME.
+
+    Polarion:
+        assignee: jhenner
+        initialEstimate: 1/20h
+        caseimportance: high
+    """
+    provider_with_special_characters.setup()
```

Comment:
I am not sure should I add a finalizer to remove the provider.

Reply:
Not necessary, the `setup_provider` fixture will enforce the `provider-limit` option and remove providers from the appliance as necessary when running any tests that run after this.